File: wscat/python/src/wscat/client/add.py
import requests
import sys
from wscat.server import simple_math_pb2

def main():
    add_req = simple_math_pb2.AddRequest(a=100, b=50)
    payload = add_req.SerializeToString()
    headers = {'Content-Type': 'application/protobuf'}

    try:
        response = requests.post("http://127.0.0.1:8000/add", data=payload, headers=headers)
        response.raise_for_status()  # Raise an exception for bad status codes

        add_resp = simple_math_pb2.AddResponse()
        add_resp.ParseFromString(response.content)
        print(f"Success! Result from server: {add_resp.result}")
    except requests.exceptions.RequestException as e:
        print(f"Error connecting to the server: {e}", file=sys.stderr)
        sys.exit(1)

if __name__ == "__main__":
    main()

File: wscat/rust/src/client/add.rs
use prost::Message;
use std::error::Error;
use wscat_rust::proto::{AddRequest, AddResponse};

#[tokio::main]
async fn main() -> Result<(), Box<dyn Error>> {
    let add_req = AddRequest { a: 100, b: 50 };
    let payload = add_req.encode_to_vec();
    
    let client = reqwest::Client::new();
    let response = client
        .post("http://127.0.0.1:8000/add")
        .header("Content-Type", "application/protobuf")
        .body(payload)
        .send()
        .await?;
    
    if !response.status().is_success() {
        eprintln!("Error: HTTP {}", response.status());
        std::process::exit(1);
    }
    
    let response_bytes = response.bytes().await?;
    let add_resp = AddResponse::decode(&response_bytes[..])?;
    
    println!("Success! Result from server: {}", add_resp.result);
    
    Ok(())
}